### PR TITLE
Fix bookmark issue and replication method for streams

### PR DESCRIPTION
**Pull request by atttiwari**
# Description of change
This PR addresses state/bookmark handling for the reports_email_activity stream (preventing null bookmarks from causing unbounded backfills) and corrects stream replication metadata so discovery reflects the stream as incremental. It also modernizes/expands the tap-tester integration suite and adds focused unit tests around the email activity archive processing and bookmark behavior. [SAC-31338](https://qlik-dev.atlassian.net/browse/SAC-31338)

Changes:

- Treat stored None bookmarks as “missing” (fallback to start_date) and avoid writing None bookmarks for empty reports_email_activity batches.
- Declare reports_email_activity as INCREMENTAL with replication key timestamp in discovery metadata, and tighten the schema so timestamp is non-null.
- Add unit tests for bookmark edge-cases + add/refresh tap-tester integration tests (discovery, sync, bookmarks, pagination, start-date, interrupted sync, field selection).

# Manual QA steps
- [x] Discovery: Running
- [x] Sync: Running
- [x] Unit tests: Running
- [x] Integration Tests: Running
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

#### AI generated code
https://internal.qlik.dev/general/ways-of-working/code-reviews/#guidelines-for-ai-generated-code
- [ ] this PR has been written with the help of GitHub Copilot or another generative AI tool

### CODE REVIEW

File: tests/unittests/test_reports_email_activity.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+def _make_client_response(operations):
+    """Return a mock client whose GET (s3=True) streams the tar.gz bytes."""
+    raw = io.BytesIO(_make_tar_gz_bytes(operations))
+    raw.read = raw.read  # already a real BytesIO
+
+    response_mock = MagicMock()
+    response_mock.raw = raw
+    response_mock.__enter__ = lambda s: s
+    response_mock.__exit__ = MagicMock(return_value=False)
```
[the remark this answers is not in the document]
Comment (atttiwari) [reply tests/unittests/test_reports_email_activity.py]: addressed
